packages/remote-feature-flag-controller/src/remote-feature-flag-controller.ts

```diff
@@ -209,9 +208,17 @@ export class RemoteFeatureFlagController extends BaseController<
     remoteFeatureFlags: FeatureFlags,
   ): Promise<FeatureFlags> {
     const processedRemoteFeatureFlags: FeatureFlags = {};
+    const metaMetricsIdResult = this.#getMetaMetricsId();
     const metaMetricsId =
-      (await this.#getMetaMetricsId) || generateFallbackMetaMetricsId();
-    const thresholdValue = generateDeterministicRandomNumber(metaMetricsId);
+      metaMetricsIdResult instanceof Promise
+        ? await metaMetricsIdResult
+        : metaMetricsIdResult;
+
+    const clientType = this.#clientConfigApiService.getClient();
+    const thresholdValue = generateDeterministicRandomNumber(
+      clientType,
+      metaMetricsId,
+    );
```

Comment:
Q: i don't really understand what you did here.
Can you elaborate more on this part please? Is it only to force the return type not to be optional or undefined as the MetaMetrics getMetaMetricsId always return a string even if the signature says it can return undefined?
What would you think about modifying the signature of MetaMetrics.getMetaMetricsId to drop undefined?

Reply:
Yes definitely! We no longer handle undefined values - the signature doesn't actually include undefined. It's about handling both synchronous and asynchronous returns in a type-safe way. 
About modifying MetaMetrics.getMetaMetricsId - it is defined in client side while mobile is async and extension is sync, and we will rely on client side to always give a value back 🙏🏻